File: sandbox-runner/patterns/sliding-window/go/starter.go
package main

import (
	"context"
	"fmt"
	"log"
	"time"

	"go.temporal.io/sdk/client"
)

func main() {
	c, err := client.Dial(client.Options{HostPort: "localhost:7233"})
	if err != nil {
		log.Fatalln("Unable to create client:", err)
	}
	defer c.Close()

	workflowID := fmt.Sprintf("%s-%d", WorkflowIDPrefix, time.Now().UnixMilli())
	we, err := c.ExecuteWorkflow(
		context.Background(),
		client.StartWorkflowOptions{
			ID:        workflowID,
			TaskQueue: TaskQueue,
		},
		SlidingWindowWorkflow,
		SlidingWindowInput{
			RecordIDs:  RecordIDs,
			WindowSize: WindowSize,
		},
	)
	if err != nil {
		log.Fatalln("Unable to execute workflow:", err)
	}
	fmt.Printf("Started workflow: %s\n", we.GetID())
	fmt.Printf("Processing %d records with window size %d…\n", len(RecordIDs), WindowSize)

	var totalProcessed int
	if err := we.Get(context.Background(), &totalProcessed); err != nil {
		log.Fatalln("Workflow failed:", err)
	}
	fmt.Printf("Sliding window complete: processed %d records\n", totalProcessed)
	fmt.Printf(
		"Open the Temporal UI and search for '%s' to see the parent and child workflows.\n",
		workflowID,
	)
}

File: sandbox-runner/patterns/sliding-window/java/src/main/java/Starter.java
import io.temporal.client.WorkflowClient;
import io.temporal.client.WorkflowOptions;
import io.temporal.serviceclient.WorkflowServiceStubs;

import java.util.HashSet;

public class Starter {
    public static void main(String[] args) {
        WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
        WorkflowClient client = WorkflowClient.newInstance(service);

        String workflowId = Shared.WORKFLOW_ID_PREFIX + "-" + System.currentTimeMillis();
        SlidingWindowWorkflow.Parent workflow = client.newWorkflowStub(
                SlidingWindowWorkflow.Parent.class,
                WorkflowOptions.newBuilder()
                        .setTaskQueue(Shared.TASK_QUEUE)
                        .setWorkflowId(workflowId)
                        .build());

        System.out.println("Started workflow: " + workflowId);
        System.out.println("Processing " + Shared.RECORD_IDS.size()
                + " records with window size " + Shared.WINDOW_SIZE + "…");
        int totalProcessed = workflow.run(
                new Shared.SlidingWindowInput(Shared.RECORD_IDS, Shared.WINDOW_SIZE, 0, 0, new HashSet<>()));
        System.out.println("Sliding window complete: processed " + totalProcessed + " records");
        System.out.println(
                "Open the Temporal UI and search for '" + workflowId
                        + "' to see the parent and child workflows.");
    }
}
